CMakeLists.txt

```diff
@@ -54,6 +54,10 @@ if (USE_CCACHE)
 else()
 	message(STATUS "ccache deselected")
 endif()
+option (USE_COMPIL_PROFILER "Use compilation profiler (for CLang >= 9 only)" OFF)
```

Comment:
I would check the actual compiler in use e.g. `CMAKE_CXX_COMPILER_ID MATCHES "Clang"` and print `FATAL_ERROR` if it doesn't meet the requirements.

Also `USE_COMPILATION_TIME_PROFILER` sounds better to me. Kinda subjective though.

Reply:
Makes sense. Done!